testutils/infra/device.py

```diff
@@ -100,6 +102,135 @@ def ssh_is_opened(self, wait=60 * 60):
         """
         self.run("true", hide=True, wait=wait)
 
+    def yocto_id_installed_on_machine(self):
+        cmd = "mender -show-artifact"
+        output = self.run(cmd, hide=True).strip()
+        return output
+
+    def get_active_partition(self):
+        cmd = "mount | awk '/on \/ / { print $1}'"
+        active = self.run(cmd, hide=True)
+        return active.strip()
+
+    def get_passive_partition(self):
+        active = self.get_active_partition()
+        cmd = (
+            "fdisk -l | grep $(blockdev --getsz %s) | grep -v %s | awk '{ print $1}'"
+            % (active, active)
+        )
+        passive = self.run(cmd, hide=True)
+        return passive.strip()
+
+    def get_reboot_detector(self, host_ip):
+        return RebootDetector(self, host_ip)
+
+
+class RebootDetector:
+    server = None
+    device = None
+    host_ip = None
+    # This global one is used to increment each port used.
+    port = 8181
+
+    def __init__(self, device, host_ip):
+        self.port = RebootDetector.port
+        RebootDetector.port += 1
+        self.host_ip = host_ip
```

Comment:
Question:
Why have a class variable, and an object variable named `host_ip` ?
Same goes for `device`

Reply:
Answer: it was like this already and I didn't dear to touch it at the time.

But yeah, I will fix it now :)